app/src/main/java/pl/example/android/garbageapp/ui/ActivitySectorGreen.java

```diff
@@ -37,6 +44,11 @@ protected void onCreate(Bundle savedInstanceState) {
         if(isMarkedForNotification())
             binding.notificationSwitch.setChecked(true);
 
-        //TODO mViewModel observe and bind data to UI
+    }
+
+    @Override
+    protected void bindDataToUI(List<SectorTerm> sectorTerms) {
+        SectorTermsAdapter adapter = (SectorTermsAdapter) binding.rcSectorTerms.getAdapter();
+        adapter.setData(sectorTerms);
```

Comment:
I would change here the code a little bit. First, move this method (along with implementation) to super class.
Then in super class declare:

`protected Binding getBinding();`

in subclasses you will need to override these methods like:

`protected Binding getBinding() {
    return DataBindingUtil.setContentView(this, R.layout.activity_sector_***);
};
`

In onCreate() method you will assign: 

`binding = getBinding();`

I think this will look better as currently bindDataToUI() methods is exactly the same in all subclasses.